src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -196,18 +197,9 @@ permutation_matrix(F::Ring, p::PermGroupElem) = permutation_matrix(F, listperm(p
 
 """
     evaluate(f::PolyElem, X::MatElem)
-
 Evaluate the polynomial `f` in the matrix `X`.
 """
-function evaluate(f::PolyElem, X::MatElem)
-   B = zero(X)
-   C = one(X)
-   for i in 0:degree(f)
-      B += C*base_ring(X)(coeff(f,i))
-      C *= X
-   end
-   return B
-end
+evaluate(f::PolyElem, X::MatElem) = sum([X^i*base_ring(X)(coeff(f,i)) for i in 0:degree(f)])
```

Comment:
that seems to replace the optimized version with the old...  but didn't we plan to just get rid of this completely? maybe that was in the other PR?

Reply:
I'll remove that in the branch `jordan-struct` (that will be merged before this one and is the one that uses the method `evaluate`).

Reply:
That branch is now merged, so now this PR needs to be rebased.